txclib/utils.py

```diff
@@ -467,8 +493,17 @@ def files_in_project(curpath):
             del dirs[:]
             continue
 
-        for f in files:
-            yield os.path.realpath(os.path.join(root, f))
+        for file_path in files:
+            full_path = os.path.realpath(os.path.join(root, file_path))
+            match = expression_regex.match(posix_path(full_path))
+            if match:
+                try:
+                    lang = match.group(1)
+                except IndexError:
+                     msg = ("File filter '{}' does not contain the '<lang>' "
+                            "placeholder".format(expression))
+                     raise Exception(msg)
```

Comment:
Why not keep the MalformedConfigFile exception here? Also you could add this case in the tests as well.

Reply:
@SofiaMargariti I kept it after all. It doesn't change anything to the message printed to the user, but let's keep in order to be able to have better error handling. I added a test as well